### PR TITLE
Support graceful restart & zero downtime

### PR DESCRIPTION
**Pull request by appleboy**
ref: https://github.com/facebookgo/grace

### CODE REVIEW

**Comment by bradrydzewski** [comment]
this is coming built-in to the Go standard library in 1.8 which should be released in a just a couple of weeks. I'm open to switching to the beta to use it:
https://beta.golang.org/doc/go1.8#http_shutdown

File: drone/server.go [hunk 1]
```diff
@@ -302,17 +303,32 @@ func server(c *cli.Context) error {
...
-			handler,
-		)
+		var err error
+
+		config := &tls.Config{
+			MinVersion: tls.VersionTLS10,
+		}
+
+		if config.NextProtos == nil {
+			config.NextProtos = []string{"http/1.1"}
```
Comment (tboerger) [inline drone/server.go]: This is the default :)

**Comment by appleboy** [comment]
1.8 support graceful shutdown but facebookgo/grace support graceful restart.

**Comment by xordspar0** [comment]
Does this PR implement graceful restarts or just shutdowns? Are we interested in both?

**Comment by ashwilliams1** [comment]
closing because `facebook/grace` has been archived and is no longer being actively maintained. Drone also added support for graceful shutdown in the 1.0 release. Perhaps Go will be adding graceful restarts to the stdlib?

**Pull request closed without merging**